cnf-certification-test/lifecycle/scaling/scaling.go

```diff
@@ -27,96 +27,152 @@ import (
+//nolint:funlen
+func runPodsetScale(podSetClient interface{}, name,
+	namespace, podsettype string, replicas int32, timeout time.Duration) error {
+	var errors error
+	switch v := podSetClient.(type) {
+	case appv1client.DeploymentInterface:
+		logrus.Infof("type is %v", v)
+		dpClient, _ := podSetClient.(appv1client.DeploymentInterface)
 		dp, err := dpClient.Get(context.TODO(), name, v1machinery.GetOptions{})
 		if err != nil {
 			logrus.Error("failed to get latest version of deployment ", namespace, ":", name)
 			return err
 		}
 		dp.Spec.Replicas = &replicas
-		_, err = clients.AppsClients.Deployments(namespace).Update(context.TODO(), dp, v1machinery.UpdateOptions{})
+		_, errors = dpClient.Update(context.TODO(), dp, v1machinery.UpdateOptions{})
+	case appv1client.StatefulSetInterface:
+		stClient := podSetClient.(appv1client.StatefulSetInterface)
+		st, err := stClient.Get(context.TODO(), name, v1machinery.GetOptions{})
 		if err != nil {
-			logrus.Error("can't update deployment ", namespace, ":", name)
+			logrus.Error("failed to get latest version of deployment ", namespace, ":", name)
 			return err
 		}
-		if !podsets.WaitForDeploymentSetReady(namespace, name, timeout) {
-			logrus.Error("can't update deployment ", namespace, ":", name)
-			return errors.New("can't update deployment")
-		}
-		return nil
-	})
-	if retryErr != nil {
-		logrus.Error("can't scale deployment ", namespace, ":", name, " error=", retryErr)
-		return false
+		st.Spec.Replicas = &replicas
+		_, errors = stClient.Update(context.TODO(), st, v1machinery.UpdateOptions{})
 	}
-	return true
+	if errors != nil {
+		logrus.Error("can't update", podsettype, " ", namespace, ":", name)
+		return errors
+	}
+
+	if !podsets.IsPodsetReady(namespace, name, podsettype, timeout) {
+		logrus.Error("can't update", podsettype, " ", namespace, ":", name)
+		return fmt.Errorf("can't update %s", podsettype)
+	}
+	return nil
 }
 
-func TestScaleHpaDeployment(deployment *v1app.Deployment, hpa *v1autoscaling.HorizontalPodAutoscaler, timeout time.Duration) bool {
+//nolint:funlen
+func TestScaleHpaDeployment(podsetlist interface{}, hpa *v1autoscaling.HorizontalPodAutoscaler, timeout time.Duration) bool {
 	clients := clientsholder.GetClientsHolder()
 	hpaName := hpa.Name
-	name, namespace := deployment.Name, deployment.Namespace
+	var name, namespace, typeset string
+	switch v := podsetlist.(type) {
+	case *v1app.StatefulSet:
+		typeset = statefulset
+		logrus.Infof("*v1app.Deployment:%v", v.Name)
+		podsetlist, _ := podsetlist.(*v1app.StatefulSet)
+		name, namespace = podsetlist.Name, podsetlist.Namespace
+	case *v1app.Deployment:
+		typeset = deployment
+		logrus.Infof("*v1app.Deployment:%v", v.Name)
+		podsetlist, _ := podsetlist.(*v1app.Deployment)
+		name, namespace = podsetlist.Name, podsetlist.Namespace
+	default:
+		fmt.Println("unknown")
+	}
+
 	hpscaler := clients.K8sClient.AutoscalingV1().HorizontalPodAutoscalers(namespace)
 	var min int32
 	if hpa.Spec.MinReplicas != nil {
```

Comment:
I think this whole section below is too complicated to not have a unit test for.

Reply:
i added the files for the test i will creat them on other PR